index.d.ts

```diff
@@ -0,0 +1,119 @@
+// Type definitions for arcsecond
+// Project: http://https://github.com/francisrstokes/arcsecond
+/* eslint-disable */
+declare namespace Arcsecond {
+
+    type ParserState<TResult = null, TData = null> = {
+        isError: boolean;
+        error: string | null;
+        target: string;
+        data: TData;
+        index: number;
+        result: TResult;
+    };
+
+    type SuccessState<TResult, TData> = ParserState<TResult, TData> & { isError: false };
+    type ErrorState<TResult, TData> = ParserState<TResult, TData> & { isError: true };
+
+    type StateData<TResult, TData> = {
+        result: TResult;
+        data: TData;
+    };
+
+    type ChainCallback<TInput, TOutput> = (result: TInput) => Parser<TOutput>;
+    type ChainFromDataCallback<TInput, TOutput, TData> = (state: StateData<TInput, TData>) => Parser<TOutput, TData>;
+    type ErrorChainCallback<TInput, TOutput, TData> = (state: ErrorState<TInput, TData>) => Parser<TOutput, TData>;
+    type ErrorMapCallback = (error: string, index: number, data) => string;
```

Comment:
`type ErrorMapCallback<TData> = (error: string, index: number, data: TData) => string;`

and in the Parser interface:

`errorMap(fn: ErrorMapCallback<TData>): Parser<TResult, TData>;`